src/frontend/src/pages/FinancePage/ReimbursementRequestsSection.tsx

```diff
@@ -33,28 +43,152 @@ const ReimbursementRequestTable = ({
+    if (b.firstName < a.firstName) {
+      return -1;
+    }
+    if (b.firstName > a.firstName) {
+      return 1;
+    }
+    return 0;
+  };
+
+  const rows = displayedReimbursementRequests.map(createReimbursementRequestRowData).sort((a, b) => {
+    if (orderBy === 'vendor') {
+      return !isAscendingOrder
+        ? vendorDescendingComparator(a.vendor, b.vendor)
+        : -vendorDescendingComparator(a.vendor, b.vendor);
+    }
+    if (orderBy === 'status') {
+      return !isAscendingOrder
+        ? statusDescendingComparator(a.status, b.status)
+        : -statusDescendingComparator(a.status, b.status);
+    }
+    if (orderBy === 'submitter') {
+      return !isAscendingOrder
+        ? submitterDescendingComparator(a.submitter, b.submitter)
+        : -submitterDescendingComparator(a.submitter, b.submitter);
+    }
+    if (b[orderBy] === undefined) {
+      return -1;
+    }
+    return !isAscendingOrder ? descendingComparator(a, b, orderBy) : -descendingComparator(a, b, orderBy);
+  });
 
   const tabs = [{ label: 'My Requests', value: 0 }];
   if (canViewAllReimbursementRequests) tabs.push({ label: 'All Club Requests', value: 1 });
 
+  const handleRequestSort = (property: keyof ReimbursementRequestRow) => {
+    setAscendingOrder(!isAscendingOrder);
+    setOrderBy(property);
+  };
+
+  const headCells: readonly ReimbursementTableHeadCell[] = [
+    {
+      id: 'submitter',
+      label: 'Recipient'
+    },
+    {
+      id: 'saboId',
+      label: 'Sabo ID'
+    },
+    {
+      id: 'amount',
+      label: 'Amount ($)'
+    },
+    {
+      id: 'dateSubmitted',
+      label: 'Date Submitted'
+    },
+    {
+      id: 'dateSubmittedToSabo',
+      label: 'Date Submitted To Sabo'
+    },
+    {
+      id: 'vendor',
+      label: 'Vendor'
+    },
+    {
+      id: 'refundSource',
+      label: 'Refund Source'
+    },
+    {
+      id: 'status',
+      label: 'Status'
+    }
+  ];
+
   return (
     <Box sx={{ bgcolor: theme.palette.background.default, width: '100%', borderRadius: '8px 8px 0 0' }}>
       <FinanceTabs tabValue={tabValue} setTabValue={setTabValue} tabs={tabs} />
       <TableContainer component={Paper} sx={{ borderRadius: '0 0 8px 8px' }}>
         <Table aria-label="simple table">
           <TableHead>
             <TableRow>
-              {tabValue === 1 && <ColumnHeader title="Recipient" />}
-              <ColumnHeader title="Sabo ID" />
-              <ColumnHeader title="Amount ($)" />
-              <ColumnHeader title="Date Submitted" />
-              <ColumnHeader title="Date Submitted To Sabo" />
-              <ColumnHeader title="Vendor" />
-              {tabValue === 1 && <ColumnHeader title="Refund Source" />}
-              <ColumnHeader title="Status" />
+              {headCells.map((headCell, i) => {
+                if (tabValue === 1 || (headCell.id !== 'submitter' && headCell.id !== 'refundSource')) {
+                  return (
+                    <TableCell
```

Comment:
I think it'd be better to incorporate the tableSortLabel and associated id/ onClick code into our ColumnHeader component

Reply:
Hi Reid, I've attempted to add the code into the column header component, but the isAscendingOrder and orderBy useStates are being modified by the tableSortLabel and read by the .sort on line 99. I'm not too sure if I would be able to transfer the isAscendingOrder and orderBy useStates from different files (from the ColumnHeader component and the ReimbursementTable component). In other words, I don't know how to transfer useStates between files (and the only other option would be to pretty much move everything from ReimbursementRequestSection into another file which also wouldn't make much sense). In addition, the last time I've tried modifying the ColumnHeader component, it caused other errors since there are other components that use the ColumnHeader component.

Reply:
you should be able to pass down the value and the setter value from each useState to the ColumnHeader component.  You could also have the ColumnHeader component conditionally return two types of components, one if those values are all passed in, and one if they're not (the one it returns now), so it doesn't break other things

Reply:
The main reason I wanted the map changed tho, is because this is happening on smaller screens and switching to column headers would fix this and IMO make the code a bit cleaner, if you can fix this issue we can leave the code more as is tho:
<img width="609" alt="image" src="https://github.com/Northeastern-Electric-Racing/FinishLine/assets/29521172/e3059f66-995d-4108-95c1-d1c4a7c1d84b">

Reply:
Sorry what is the issue in the screenshot? Is it cutting off?